### DISCUSSION TITLE
Workflow HTTP Request relative path?

### DISCUSSION THREAD

**Discussion by alXDraghiciu** [OrchardCMS/OrchardCore#11325] [Q&A] [answered]
Hi. How does accessing a relative path from my own domain in an HTTP Request inside a workflow work? 

The description says it supports Liquid syntax but _Request.Host_ does not work, neither does _~_ operator. I always get the message:

> An invalid request URI was provided. Either the request URI must be an absolute URI or BaseAddress must be set.

I've tried setting BaseUrl from config but still same error. 

How can I access my domain URL and make the relative request there?

**Comment by alXDraghiciu** [comment]
I've also tried `absolute_url` Liquid filter to no results. It does not seem to work at all inside HTTPRequest Activity URL, even simply {{ '~/' | absolute_url }} will not work. It should return the tenant URL right?

**Reply by Skrypt** [reply]
https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#site

{{ Site.BaseUrl }}

Gives you the current tenant BaseUrl. Though that's probably not what you are looking for.

Else you can form the URL using the Request object directly : 

https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#site

So something like : 

{{ Request.Scheme }}://{{ Request.Host }}

**Comment by jtkech** [comment, marked as answer]
@alXDraghiciu 

You need to use the `raw` filter to prevent Liquid to html encode the URI, just tried the following that works

    {{ Site.BaseUrl | raw }}

    {{ "/about" | absolute_url | raw }}

**Updated**

Also as stated by @Skrypt you can use the `Request` accessor, still with the raw filter, here applied on the whole

    {{ Request.Scheme | append:"://" | append: Request.Host | raw }}

Or this pattern where anyway we need the raw filter applied on the `://` separator, but notice the usage of `UriHost` that correctly uri encode this component and return a liquid string value marked to not be html encoded (see below).

    {{ Request.Scheme }}{{ "://" | raw }}{{ Request.UriHost }}

For info

https://github.com/OrchardCMS/OrchardCore/blob/275ce1e6228c52016196a22163953470dcabe0b9/src/OrchardCore/OrchardCore.DisplayManagement.Liquid/OrchardCoreBuilderExtensions.cs#L106-L112

**Finally**: To be tenant aware

    {{ Request.Scheme }}{{ "://" | raw }}{{ Request.UriHost }}{{ Request.UriPathBase }}

**Reply by alXDraghiciu** [reply]
Thank you that worked flawlessly.
